scalyr_agent/configuration.py

```diff
@@ -1035,6 +1093,80 @@ def __verify_main_config_and_apply_defaults(self, config, file_path, apply_defau
+
+        If the environment value is returned, it is also written into the config_object.
+
+        Currently only handles the following types (str, int, bool, float, JsonObject, JsonArray).
+        Also validates that environment variables can be correctly converted into the primitive type.
+
+        If ENVIRONMENT_AWARE_CONFIG_PARAMS map does not contain a value for the field, the environment
+        variable name is derived by prefixing 'SCALYR_' to the field name. Otherwise, the map value is used verbatim as
+        the environment variable name.
+
+        Both upper-case and lower-case versions of the environment variable will be checked.
+
+        @param config_object: The JsonObject config containing the field as a key
+        @param param_name: Parameter name
+        @param param_type: Parameter primitive type
+        @return A python object representing the config param (or environment) value or None
+        @raises
+            JsonConversionException: if the config value or env value cannot be correctly converted.
+            TypeError: if the param_type is not supported.
+        """
+        if param_type == int:
+            config_val = config_object.get_int(param_name, none_if_missing=True)
+        elif param_type == bool:
+            config_val = config_object.get_bool(param_name, none_if_missing=True)
+        elif param_type == float:
+            config_val = config_object.get_float(param_name, none_if_missing=True)
+        elif param_type == str:
+            config_val = config_object.get_string(param_name, none_if_missing=True)
+        elif param_type == JsonObject:
+            config_val = config_object.get_json_object(param_name, none_if_missing=True)
+        elif param_type == JsonArray:
+            config_val = config_object.get_json_array(param_name, none_if_missing=True)
+        elif param_type == ArrayOfStrings:
+            # ArrayOfStrings are extracted from config file as JsonArray
+            # (but extracted from the environment different from JsonArray)
+            config_val = config_object.get_json_array(param_name, none_if_missing=True)
+        else:
+            raise TypeError('Unsupported environment variable conversion type %s (param name = %s)'
+                            % (param_type, param_name))
+
+        if param_name not in Configuration.ENVIRONMENT_AWARE_CONFIG_PARAMS:
+            return config_val
+
+        env_name, env_str, env_val = scalyr_util.get_config_from_env(
+            param_name,
+            custom_env_name=Configuration.ENVIRONMENT_AWARE_CONFIG_PARAMS.get(param_name),
+            convert_to=param_type,
+        )
+
+        # Not set in environment
+        if env_val is None:
+            return config_val
+
+        # Report conflicting values
+        if self.__logger:
+            if config_val is not None and config_val != env_val:
+                self.__logger.warn(
```

Comment:
Because we check config once every 30 seconds, we would be logging this every 30 seconds.  Use the `limit_once_per_x_secs`  logging features to emit this once every 5 minutes.

Reply:
will fix